nb-api/src/main/java/io/nosqlbench/components/NBBaseComponent.java

```diff
@@ -122,6 +126,14 @@ public final void close() throws RuntimeException {
         }
     }
 
+    public void reportExecutionMetric(NBMetric m) {
+        if (parent != null) {
+            parent.reportExecutionMetric(m);
+        } else {
+            logger.warn(() -> "Component " + this.toString() + " has a non-session null parent component");
+        }
+    }
+
```

Comment:
This is perhaps a dumb question, apologies in advance. 

Each `NBComponent` has a single parent `NBComponent`, and the root of the tree corresponds to the Session, correct? 

If so, this is baiscally just parenting up to the root/Session, and once we reach the root/Session, we just emit the warning message?

Reply:
Never mind, just realized we override this method in `NBSession`.